extension/data_loader/mmap_data_loader.cpp

```diff
@@ -63,14 +62,16 @@ MmapDataLoader::~MmapDataLoader() {
   std::free(const_cast<char*>(file_name_));
   // fd_ can be -1 if this instance was moved from, but closing a negative fd is
   // safe (though it will return an error).
-  ::close(fd_);
+  if (fd_ != -1) {
+    ::close(fd_);
```

Comment:
I'm surprised this (and open, etc.) works, since it was the problem in #8913.  I haven't built this file on Windows myself yet, but maybe a similar fix is necessary?